### ISSUE TITLE
Rest export for Comulis to be corrected

### ISSUE THREAD

**Issue by PerrineGilloteaux** [NEUBIAS/bise#209]
Following the update, the  Rest export view of serach need to be checked since 
https://www.comulis.eu/correlation-software#search_result will return Failed retrieving data from biii! 0 -

See also #126

**Comment by PerrineGilloteaux** [comment]
It is actually the return of the CORS error #113 . 
ps://biii.eu/all-content-rest?type=software&field_has_topic_target_id_1%5B%5D=4774&_format=json&source=COMULIS' from origin 'https://www.comulis.eu' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by PerrineGilloteaux** [comment]
Access-Control-Allow-Origin * in .htacess

**Comment by miura** [comment]
Looks ok now?
https://biii.eu/all-content-rest?type=software&field_has_topic_target_id_1%5B%5D=4774&_format=json&source=COMULIS

**Comment by PerrineGilloteaux** [comment]
Hello, to test you need to go to the comulis software https://www.comulis.eu/correlation-software#search_result but first edit the .htaccess from the server directly ( and add it to .git ignore if it works) to remise the cors error. i think i had no server error this morning ( on prod but i had on gitpod). I will try tomorrow early morning otherwise. We also need to test the fiji search bar which may have the same problem.

**Comment by miura** [comment]
@PerrineGilloteaux 

You probably mean this one in 2019 right?

https://github.com/NEUBIAS/bise/pull/114/files#diff-270939b4fba4be968ab78e23dc0207eb893744491980a25c99a27c809a82ddab

In fact, I have never applied this change as it crashed the whole biii.eu (production). See:

https://github.com/NEUBIAS/bise/pull/114

I guess it's better to set the proper headers for COR as it is explained in the link you found.

https://drupal.stackexchange.com/questions/312414/cors-policy-no-access-control-allow-origin

In biii, service.yml is bise.service.yml. 
Here:

https://github.com/NEUBIAS/bise/blob/dev/web/sites/default/bise.services.yml

In the `allowdHeader` array, the first item is missing a single colon. 
Maybe try fixing this? It's possible that with Drupal9, the parsing has become more strict. Please also check the counterpart service.yml file. 

![image](https://github.com/NEUBIAS/bise/assets/272781/1dea3f81-eeff-45a7-8e81-2abe21e8a6b5)

**Comment by joakimlindblad** [comment]
Currently the COMULIS search **does not work** due to the current CORS policy of the Biii-site, which does not allow external sites to ask for BIII resources. 

@miura , as commented by @PerrineGilloteaux above, to check if it works you just need to go to https://www.comulis.eu/correlation-software#search_result and click any of the three buttons. 

When failing, you may also see the following in the browser console (Ctrl-J in Chrome):
`correlation-software#search_result:1 Access to XMLHttpRequest at 'https://biii.eu/all-content-rest?type=software&field_has_topic_target_id_1%5B%5D=4774&_format=json&source=COMULIS' from origin 'https://www.comulis.eu' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by miura** [comment]
Hi @joakimlindblad  

Yes, I do understand the problem as the symptom is similar. 
But the reason probably is not the same (I checked all the external interfaces and should be OK) but more internal. 

I now found this error that comes up in relation to the use of php8:

<img width="1591" alt="image" src="https://github.com/NEUBIAS/bise/assets/272781/d9318042-9956-4153-a642-490569447262">

This actually happens also in some occasions with clicking internal links. 
I will try adding the patch below later. 

https://www.drupal.org/project/field_group/issues/3311905

**Comment by joakimlindblad** [comment]
Yes, in deed now the error looks completely different; presumably after you added the mentioned .htaccess line.
I will check back once the relevant looking patch is applied.

**Comment by PerrineGilloteaux** [comment]
You are my hero @miura

**Comment by miura** [comment]
@PerrineGilloteaux @albangaignard 
So my best bet is that this is now properly fixed with this commit. 
https://github.com/NEUBIAS/bise/commit/a00d8e3e01b27a065bcc4d182d6af602c1b67aa0

This fix is in the production site already.
